common/src/main/java/org/opensearch/ml/common/connector/HttpConnector.java

```diff
@@ -377,17 +379,32 @@ private List<String> findStringParametersWithNullDefaultValue(String input) {
     }
 
     @Override
-    public void decrypt(String action, BiFunction<String, String, String> function, String tenantId) {
-        Map<String, String> decrypted = new HashMap<>();
+    public void decrypt(String action, BiFunction<String, String, Future<String>> function, String tenantId) {
+        Map<String, Future<String>> decryptingTempCredential = new HashMap<>();
+        decryptedCredential = new HashMap<>();
         for (String key : credential.keySet()) {
-            decrypted.put(key, function.apply(credential.get(key), tenantId));
+            decryptingTempCredential.put(key, function.apply(credential.get(key), tenantId));
         }
-        this.decryptedCredential = decrypted;
+        fillCredential(decryptingTempCredential, decryptedCredential);
         Optional<ConnectorAction> connectorAction = findAction(action);
         Map<String, String> headers = connectorAction.map(ConnectorAction::getHeaders).orElse(null);
         this.decryptedHeaders = createDecryptedHeaders(headers);
     }
 
+    private void fillCredential(Map<String, Future<String>> decrypted, Map<String, String> decryptedCredential) {
+        for (String key : decrypted.keySet()) {
+            try {
+                if (decrypted.get(key) != null) {
+                    decryptedCredential.put(key, decrypted.get(key).get());
+                } else {
+                    decryptedCredential.put(key, null);
```

Comment:
when the key is null, and you put value of null, what is the intention here? why don't you skip when the key is null?